sw/device/tests/rstmgr_cpu_info_test.c

```diff
@@ -0,0 +1,204 @@
+ *
+ */
+// ibex_mepc_read : exception addr
+// ibex_mtval_read : last_data_addr
+
+void ottf_exception_handler(void) {
+  uint32_t addr = (uint32_t)exp_dump;
+
+  temp_dump[kCpuDumpIdxCurrentExceptionAddr] =
+      (dif_rstmgr_cpu_info_dump_segment_t)ibex_mepc_read();
+  temp_dump[kCpuDumpIdxPreviousExceptionAddr] =
+      temp_dump[kCpuDumpIdxCurrentExceptionAddr];
+  temp_dump[kCpuDumpIdxCurrentLastDataAddr] = kIllegalAddr2;
+  temp_dump[kCpuDumpIdxPreviousLastDataAddr] = kIllegalAddr2;
+  temp_dump[kCpuDumpIdxPreviousPc] =
+      temp_dump[kCpuDumpIdxPreviousExceptionAddr] + 2;
+  temp_dump[kCpuDumpIdxPreviousValid] = 1;
+
+  CHECK(flash_ctrl_testutils_write(&flash_ctrl, addr, 0, temp_dump,
+                                   kDifFlashCtrlPartitionTypeData, 9) == 0);
+
+  read_error();
+}
+
+bool test_main(void) {
+  dif_rstmgr_t rstmgr;
+  dif_aon_timer_t aon_timer;
+  dif_pwrmgr_t pwrmgr;
+
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+  // Initialize flash_ctrl
+  CHECK_DIF_OK(dif_flash_ctrl_init_state(
+      &flash_ctrl,
+      mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR)));
+
+  // Enable flash access
+  flash_ctrl_testutils_default_region_access(&flash_ctrl,
+                                             /*rd_en*/ true,
+                                             /*prog_en*/ true,
+                                             /*erase_en*/ true,
+                                             /*scramble_en*/ false,
+                                             /*ecc_en*/ false,
+                                             /*he_en*/ false);
+
+  dif_rstmgr_reset_info_bitfield_t rst_info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rst_info));
+
+  if (rst_info == kDifRstmgrResetInfoPor) {
+    LOG_INFO("Booting for the first time, setting wdog");
+
+    global_error_cnt = 0;
+    uint32_t bark_cycles = aon_timer_testutils_get_aon_cycles_from_us(200);
+    uint32_t bite_cycles = aon_timer_testutils_get_aon_cycles_from_us(200);
+
+    // Set wdog as a reset source.
+    CHECK_DIF_OK(dif_pwrmgr_set_request_sources(&pwrmgr, kDifPwrmgrReqTypeReset,
+                                                kDifPwrmgrResetRequestSourceTwo,
+                                                kDifToggleEnabled));
+
+    // Setup the wdog bark and bite timeouts.
+    aon_timer_testutils_watchdog_config(&aon_timer, bark_cycles, bite_cycles,
+                                        false);
+
+    // enable cpu info
+    CHECK_DIF_OK(dif_rstmgr_cpu_info_set_enabled(&rstmgr, kDifToggleEnabled));
+    read_error();
+  } else {
+    LOG_INFO("Comes back after bite");
+    LOG_INFO("global_cnt:%0d", global_error_cnt);
```

Comment:
does this global error count still display correctly after the watchdog reset..? wouldn't the C runtime zero initialize this or something? 

@timothytrippel

Reply:
> does this global error count still display correctly after the watchdog reset..? wouldn't the C runtime zero initialize this or something?
> 
> @timothytrippel

Let me remove it, global_error_cnt only for creating double fault. After reset,
it is useless.